### PR TITLE
Bind multi range query API

### PR DESCRIPTION
**Pull request by tdenniston**

### CODE REVIEW

File: src/main/java/io/tiledb/java/api/Query.java [hunk 1]
```diff
@@ -161,6 +161,92 @@ public Query setSubarray(NativeArray subarray) throws TileDBError {
...
+   * the range components must be the same as the type of the domain of the array in the query.
+   *
+   * @param dimIdx The index of the dimension to add the range to
+   * @param start The range start
+   * @param end The range end
+   * @return This query
+   * @throws TileDBError A TileDB exception
+   */
+  public Query addRange(int dimIdx, Object start, Object end) throws TileDBError {
+    Datatype dimType = array.getSchema().getDomain().getType();
```
Comment (jakebolewski) [inline src/main/java/io/tiledb/java/api/Query.java]: you need to wrap these in a try block as well

try (Schema schema = array.getSchema(); Domain = schema.getDomain()) {}`

File: src/main/java/io/tiledb/java/api/Query.java [hunk 2]
```diff
@@ -161,6 +161,92 @@ public Query setSubarray(NativeArray subarray) throws TileDBError {
...
+  /**
+   * Retrieves a specific range of the query subarray along a given dimension.
+   *
+   * @param dimIdx The index of the dimension to retrieve the range from
+   * @param rangeIdx The index of the range to retrieve
+   * @return Pair of (start, end) of the range.
+   * @throws TileDBError A TileDB exception
+   */
+  public Pair<Object, Object> getRange(int dimIdx, long rangeIdx) throws TileDBError {
+    Datatype dimType = array.getSchema().getDomain().getType();
```
Comment (jakebolewski) [inline src/main/java/io/tiledb/java/api/Query.java]: Same here

**Comment by tdenniston** [comment]
Thanks @jakebolewski ! Fixed those two issues

**Comment by jakebolewski** [comment]
LGTM

**Pull request merged**